src/lexer/src_string.sig

```diff
@@ -1,7 +1,9 @@
 signature SRC_STRING =
 sig
     type yypos = int
-
+    exception StringNotClosed of int * int (* line position * line number *)
+    val startPos : int ref
```

Comment:
I don't like exposing refs if we only intend to use them to access the value. This should be `val startPos : unit -> int`.